docs/tutorials/typescript/getting_started.md

```diff
@@ -0,0 +1,141 @@
+
+> You'll be provided with a URL to login with your NEAR account.
+>
+> Example:
+
+```shell
+$> nearai login
+
+Please visit the following URL to complete the login process: https://auth.near.ai?message=Welcome+to+NEAR+AI&nonce=<xyzxyzxyzxyzx>&recipient=ai.near&callbackUrl=http%3A%2F%2Flocalhost%3A63130%2Fcapture
+```
+
+- [ ] Login with NEAR by connecting with your wallet. After successfully logging in, you will see a confirmation screen. Close it and return to your terminal.
+
+## Local Development
+
+Clone the NEAR AI Github Repository
+
+```shell
+git clone https://github.com/nearai/nearai.git
+```
+
+Change your directory to `ts_runner/ts_agent` and Install `Node.js` dependencies by running the command below
+
+```shell
+cd ts_runner/ts_agent && npm install
+```
+
+Change the directory to `ts_runner/ts_agent/agents` and add your agent related code in `agent.ts` file
+
+```typescript
+import { env } from "ts-agent-runner";
+
+(async () => {
+  try {
+    const userMessage = "Tell me an AI joke";
+
+    console.log("User input:", userMessage);
+
+    const messages: any = [
+      {
+        role: "system",
+        content: "You are a smart assistant.",
+      },
+      {
+        role: "user",
+        content: userMessage,
+      },
+    ];
+
+    // inference
+    const reply = await env.completion(
+      messages,
+      "llama-v3p1-70b-instruct",
+      4000,
+      0.5
+    );
+
+    if (env.get_thread_id() !== "thread_local") {
+      await env.add_reply(reply);
+    }
+
+    console.log("Agent output:", reply);
+  } catch (error) {
+    console.error("Agent error:", error);
+  }
+})();
+```
+
+Run the agent
+
+```shell
+npm run build && npm run start agents/agent.ts
+```
+
+## Deployment
+
+> This process will be simplified in the future. For now, you can follow these steps to deploy your TypeScript agent.
+
+Create an Agent
+
+```shell
+nearai agent create
+```
+
+> For more detail follow the link: [NEAR AI Quickstart - NEAR AI](https://docs.near.ai/agents/quickstart/)
+> Name can’t contain space. Example: Agent name have to be AgentName and can’t be “Agent Name”
+
+During the agent creation process, `nearai` builds your agent in your local AI registry located at:
+
+`/home_directory/.nearai/registry/<your-account.near>/<agent-name>/0.0.1`
+
+This folder contains two files that define your agent:
+
+1. `metadata.json`: Contains information / configuration about your agent.
+2. `agent.py`: Python code that executes each time your agent receives a prompt.
+
+Update your Agent
+
+- Rename the [`agent.py`](http://agent.py) to `agent.ts` and copy-paste your agent code
+- Update the `metadata.json` file and include `"framework": "ts",` line
+
+```json
+{
+  "category": "agent",
+  ...
+  "details": {
+    "agent": {
+      "framework": "ts",
+      ...
+    }
+  },
+  ...
+}
+```
+
+Interacting with your Agent locally
+
+```shell
+# Run agent locally
+nearai agent interactive <path-to-agent> --local
```

Comment:
This did not work for me. The agent only ran successfully when in `ts_runner/ts_agent/agents`. The above flow created an agent in my local registry and threw an error when running.